sbot/metafeed_test.go

```diff
@@ -264,3 +262,170 @@ func TestMultiFeedSync(t *testing.T) {
+	idxSubfeed, err := bot.MetaFeeds.CreateSubFeed(bot.KeyPair.ID(), "indexes", refs.RefAlgoFeedBendyButt)
+	r.NoError(err)
+
+	// now create two index feeds on the just created subfeed for indexes
+	idx1, err := bot.MetaFeeds.CreateSubFeed(idxSubfeed, "index-foo", refs.RefAlgoFeedGabby)
+	r.NoError(err)
+
+	idx2, err := bot.MetaFeeds.CreateSubFeed(idxSubfeed, "index-bar", refs.RefAlgoFeedGabby)
+	r.NoError(err)
+
+	// check we can list them. 
+	// first, check that the root mf only has one feed (i.e. idxSubFeed)
+	lst, err := bot.MetaFeeds.ListSubFeeds(bot.KeyPair.ID())
+	r.NoError(err)
+	r.Len(lst, 1, "the root mf has more than one subfeed")
+	r.True(lst[0].Feed.Equal(idxSubfeed))
+
+	// then, verify that the subfeed has the two new index feeds
+	lst, err = bot.MetaFeeds.ListSubFeeds(idxSubfeed)
+	r.NoError(err)
+	r.Len(lst, 2, "more than two indexes on the subfeed")
+
+	// write contains util to make sure that the two indexes exist (not guaranteed to be ordered)
+	has := func (feedlist []ssb.SubfeedListEntry, target refs.FeedRef) bool {
+		var found bool
+		for _, entry := range feedlist {
+			if entry.Feed.Equal(idx1) {
+				found = true
+				break
+			}
+		}
+		return found
+	}
+	r.True(has(lst, idx1), "found idx1")
+	r.True(has(lst, idx2), "found idx2")
+
+	// check we can publish as idx1 and idx2
+	_, err = bot.MetaFeeds.Publish(idx1, refs.NewPost("foo!"))
+	r.NoError(err)
+	_, err = bot.MetaFeeds.Publish(idx2, refs.NewPost("bar!"))
+	r.NoError(err)
+
+	// try to tumbstone on the wrong mount
+	err = bot.MetaFeeds.TombstoneSubFeed(bot.KeyPair.ID(), idx1)
+	r.Error(err)
+
+	// listings stay the same
+	lst, err = bot.MetaFeeds.ListSubFeeds(bot.KeyPair.ID())
+	r.NoError(err)
+	r.Len(lst, 1, "the root mf has more than one subfeed after faulty tombstone")
+
+	lst, err = bot.MetaFeeds.ListSubFeeds(idxSubfeed)
+	r.NoError(err)
+	r.Len(lst, 2, "more than two indexes on the subfeed after faulty tombstone")
+
+	// now remove index from the right mount
+	err = bot.MetaFeeds.TombstoneSubFeed(idxSubfeed, idx1)
+	r.NoError(err)
+
+	lst, err = bot.MetaFeeds.ListSubFeeds(idxSubfeed)
+	r.NoError(err)
+	r.Len(lst, 1, "more subfeeds on the index?")
+
+	/* test index feed creation */
+	// initial notes from cryptix:
+	// * have a normal/base feed,
+	// * create indexes,
+	// * publish on the base,
+	// * check the index messages exist..
+
+	// create a main feed (holds actual messages, regular old ssb feed thinger) on the root metafeed
+	mainFeedRef, err := bot.MetaFeeds.CreateSubFeed(bot.KeyPair.ID(), "main", refs.RefAlgoFeedSSB1)
+	r.NoError(err)
+
+	// note (cblgh): hmm might be nice with a <sbot>.MetaFeeds.Get(subfeedRef) (subfeed, error) method?
+
+	// create two new index feeds (distinct from the test two indexes: idx1 idx2)
+	// note (cblgh): might we introduce a method <sbot>.MetaFeeds.CreateIndex? 
+	// => creates a subfeed that is an index, with the correct `query` parameter & querylang: ssb-ql-0 set
```

Comment:
this question might be answered by https://github.com/ssb-ngi-pointer/go-metafeed/pull/27 :~